app/actions/remote/custom_profile.ts

```diff
@@ -0,0 +1,145 @@
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const fields = await client.getCustomProfileAttributeFields();
+
+        if (fields?.length > 0) {
+            // Store fields in the database
+            await operator.handleCustomProfileFields({
+                fields,
+                prepareRecordsOnly: false,
+            });
+            return {fields, error: undefined};
+        }
+        return {fields: [], error: undefined};
+    } catch (error) {
+        logDebug('error on fetchCustomProfileFields', getFullErrorMessage(error));
+        forceLogoutIfNecessary(serverUrl, error);
+        return {fields: [], error};
+    }
+};
+
+/**
+ * Fetches custom profile attribute values for a user
+ * @param serverUrl - The server URL
+ * @param userId - The user ID
+ * @param filterEmpty - Whether to filter out empty values
+ * @returns Promise with the custom profile attributes or error
+ */
+export const fetchCustomProfileAttributes = async (serverUrl: string, userId: string, filterEmpty = false): Promise<{attributes: CustomAttributeSet; error: unknown}> => {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const [fields, attrValues] = await Promise.all([
+            client.getCustomProfileAttributeFields(),
+            client.getCustomProfileAttributeValues(userId),
+        ]);
+
+        if (fields?.length > 0) {
+            const attributes: Record<string, CustomAttribute> = {};
+            const attributesForDatabase: CustomProfileAttribute[] = [];
+
+            fields.forEach((field: CustomProfileField) => {
+                const value = (attrValues as Record<string, string>)[field.id] || '';
+                if (!filterEmpty || value) {
+                    attributes[field.id] = {
+                        id: field.id,
+                        name: field.name,
+                        value,
+                        sort_order: field.attrs?.sort_order,
+                    };
+
+                    // Prepare attributes for database storage
+                    attributesForDatabase.push({
+                        id: `${field.id}-${userId}`,
+                        field_id: field.id,
+                        user_id: userId,
+                        value,
+                    });
+                }
+            });
+
+            // Store attributes in the database
+            if (attributesForDatabase.length > 0) {
+                await operator.handleCustomProfileAttributes({
+                    attributes: attributesForDatabase,
+                    prepareRecordsOnly: false,
+                });
+            }
+            if (fields?.length > 0) {
+                await operator.handleCustomProfileFields({
+                    fields,
+                    prepareRecordsOnly: false,
+                });
+            }
```

Comment:
why not use `prepareRecordsOnly` as true and then batch the results? I'm not saying it needs to be done that way, just want to understand.

Reply:
I'll modify to use the prepareRecordsOnly, no especial reason for using one or the other.